nixos/modules/services/backup/borgmatic.nix

```diff
@@ -0,0 +1,247 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.borgmatic;
+  cfgfile = pkgs.writeText "config.yaml" (builtins.toJSON (recursiveUpdate {
```

Comment:
I would highly recommend reading through the [freeform module section](https://nixos.org/manual/nixos/stable/index.html#sec-freeform-modules) of the manual. You can take a look at [RFC 42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md#detailed-design) for some background on this section.